lib/Jsrt/ChakraCommon.h

```diff
@@ -2978,4 +2978,343 @@ typedef unsigned char* ChakraBytePtr;
+            _In_ JsTTDGetLogStreamCallback getLogStreamInfo,
+            _In_ JsTTDGetSnapshotStreamCallback getSnapshotStreamInfo,
+            _In_ JsTTDGetSrcCodeStreamCallback getSrcCodeStreamInfo,
+            _In_ JsTTDReadBytesFromStreamCallback readBytesFromStream,
+            _In_ JsTTDWriteBytesToStreamCallback writeBytesToStream,
+            _In_ JsTTDFlushAndCloseStreamCallback flushAndCloseStream);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Start Time-Travel Recording.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDStartTimeTravelRecording();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Stop Time-Travel Recording.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDStopTimeTravelRecording();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Emit Time-Travel Recording.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDEmitTimeTravelRecording();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Start Time-Travel Debugging.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDStartTimeTravelDebugging();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Pause Time-Travel recording gefor executing code on behalf of debugger or other diagnostic/telemetry.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDPauseTimeTravelBeforeRuntimeOperation();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     ReStart Time-Travel recording after executing code on behalf of debugger or other diagnostic/telemetry.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDReStartTimeTravelAfterRuntimeOperation();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Notify the Js runtime that the host as created/canceled a callback with the given function and id.
+    /// </summary>
+    /// <param name="isCancel">True if the action is to create the callback with the callbackId.</param>
+    /// <param name="isCancel">True if the action is to cancel the callback with the callbackId.</param>
+    /// <param name="isRepeating">True if the action is to create a repeating callback (e.g., setInterval).</param>
```

Comment:
If the host does a setInterval instead of setTimeout or a .on handler that may happen many times. We don't depend on this being set (so always setting it to false is ok) but if this info is provided we can do additional diagnostics.